lib/tests/test_core.c

```diff
@@ -130,6 +130,49 @@ test_tskit_version(void)
     CU_ASSERT_EQUAL(TSK_VERSION_PATCH, 15);
 }
 
+static void
+test_gsl_ran_flat_patch(void)
+{
+    double left, right, x;
+    int i, ret, child_status;
+    gsl_rng *rng = gsl_rng_alloc(gsl_rng_default);
+    pid_t pid = fork();
+
+    right = 85190021.000000;
+    left = nexttoward(right, 0.0);
+    gsl_rng_set(rng, 12345);
+    x = gsl_ran_flat(rng, left, right);
+    CU_ASSERT(!(right > x));
+    gsl_rng_set(rng, 12345);
+    x = msp_gsl_ran_flat(rng, left, right);
+    CU_ASSERT(right > x);
+    CU_ASSERT(left <= x);
+
+    left = 0.0;
+    right = 1.0;
+    for (i = 0; i < 1000; i++) {
+        x = msp_gsl_ran_flat(rng, left, right);
+        CU_ASSERT(left <= x);
+        CU_ASSERT(right > x);
+    }
+
+    left = 2.0;
+    right = 2.0;
+    x = msp_gsl_ran_flat(rng, left, right);
+    CU_ASSERT(left == x && right == x);
+
+    left = 3.0;
```

Comment:
Ooh, clever! I think this might end up being a bit brittle though, so maybe not worth the complication? I wouldn't bother personally, we're not going to remove the assert from the msp_gsl_ran_flat function and that's all this is testing essentially.